buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -295,11 +295,22 @@ private boolean tryExpandMagazines(int currentLength) {
                 if (mags.length >= MAX_STRIPES || mags.length > currentLength) {
                     return true;
                 }
-                Magazine[] expanded = Arrays.copyOf(mags, mags.length * 2);
-                for (int i = mags.length, m = expanded.length; i < m; i++) {
+                int preferredChunkSize = mags[0].sharedPrefChunkSize;
+                Magazine[] expanded = new Magazine[mags.length * 2];
+                for (int i = 0, m = expanded.length; i < m; i++) {
                     expanded[i] = new Magazine(this);
+                    expanded[i].localPrefChunkSize = preferredChunkSize;
+                    expanded[i].sharedPrefChunkSize = preferredChunkSize;
                 }
                 magazines = expanded;
+                for (Magazine magazine : mags) {
+                    long stamp = magazine.allocationLock.writeLock();
+                    try {
+                        magazine.free();
```

Comment:
I would move the locking to the `free()` method that is also consistent with what we do in `tryAllocate(...)`